Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -73,15 +87,31 @@ protected override void Started(EntityUid uid, RevolutionaryRuleComponent compon
     protected override void ActiveTick(EntityUid uid, RevolutionaryRuleComponent component, GameRuleComponent gameRule, float frameTime)
     {
         base.ActiveTick(uid, component, gameRule, frameTime);
+        // Check for command loss or if a banned player needs to be smited.
         if (component.CommandCheck <= _timing.CurTime)
         {
             component.CommandCheck = _timing.CurTime + component.TimerWait;
 
+            // Check for command loss
             if (CheckCommandLose())
             {
                 _roundEnd.DoRoundEndBehavior(RoundEndBehavior.ShuttleCall, component.ShuttleCallTime);
                 GameTicker.EndGameRule(uid, gameRule);
             }
+
+            // Execute scheduled smites for banned players
+            if (_scheduledSmites.Count > 0)
+            {
+                var currentTime = _timing.CurTime;
+                foreach (var entity in _scheduledSmites.Keys.ToList().Where(entity => _scheduledSmites[entity] <= currentTime))
+                {
+                    if (EntityManager.EntityExists(entity))
+                    {
+                        _adminVerbSystem.RandomDeath(entity);
+                    }
+                    _scheduledSmites.Remove(entity);
+                }
+            }
         }
```

Comment:
I'm not sure if this is better or worse than a timer but at least revs is getting bulldozed someday.